test/testsuites/clusteroperatorstatustests.go

```diff
@@ -0,0 +1,57 @@
+package testsuites
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/operator-framework/operator-sdk/pkg/test"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// ClusterOperatorStatusOnStartup is a test suite that ensures the ClusterOperator resource which
+// defines the status of the marketplace operator has the correct status upon initialization
+func ClusterOperatorStatusOnStartup(t *testing.T) {
+	ctx := test.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	// Get global framework variables.
+	client := test.Global.Client
+
+	// Get namespace
+	namespace, err := test.NewTestCtx(t).GetNamespace()
+	if err != nil {
+		t.Fatalf("Could not get namespace: %v", err)
+	}
+
+	// Check that the ClusterOperator resource has the correct status
+	clusterOperatorName := "marketplace"
+	expectedTypeStatus := map[configv1.ClusterStatusConditionType]configv1.ConditionStatus{
+		"Progressing": "False",
+		"Available":   "True",
+		"Degraded":    "False"}
+	err = clusterOperatorCheckTypeStatus(client, clusterOperatorName, namespace, expectedTypeStatus)
+	if err != nil {
+		t.Fatalf("ClusterOperator never reached expected status: %v", err)
+	}
+}
+
+// clusterOperatorCheckTypeStatus checks each condition (type & status) against expectedTypeStatus
+func clusterOperatorCheckTypeStatus(client test.FrameworkClient, clusterOperatorName string, namespace string, expectedTypeStatus map[configv1.ClusterStatusConditionType]configv1.ConditionStatus) error {
+	// Get ClusterOperator resource with clusterOperatorName name
+	namespacedName := types.NamespacedName{Name: clusterOperatorName, Namespace: namespace}
+	result := &configv1.ClusterOperator{}
+	// No need to poll here since the ClusterOperator resource should be there when tests are launched
```

Comment:
Why shouldn't you poll? It is possible the tests run and for some reason the operator is coming up slowly on the cluster.

Reply:
ok I agree - what timeout length would you advise here?

Reply:
I would say at least 5-10 minutes. But others should chime in here too.

Reply:
Thanks! @awgreene @kevinrizza @anik120 what do you think?